gui/src/3D/controls/control3D.ts

```diff
@@ -402,7 +403,11 @@ export class Control3D implements IDisposable, IBehaviorAware<Control3D> {
     }
 
     /** @hidden */
-    public _processObservables(type: number, pickedPoint: Vector3, pointerId: number, buttonIndex: number): boolean {
+    public _processObservables(type: number, pickedPoint: Vector3, originMeshPosition: Nullable<Vector3>, pointerId: number, buttonIndex: number): boolean {
+        if (this instanceof TouchButton3D && originMeshPosition) {

```

Comment:
Avoid loading the class (or using instanceOf). The type can be loaded using typescript type loading (i.e.:

```javascript
declare type TouchButton3D = import("./touchButton3D").TouchButton3D;
```

and instanceOf should be replaced with some form of feature detection (for example check `_getTypeName` instead).

Reply:
_getTypeName returns the name of the class, but I want the TouchButton3D and all of its extended classes, like TouchMeshButton3D.